opam lint improvements

Thanks for `opam lint` ! 

One problem is that you are (or at least I am) likely to use `opam lint` on the `opam` file of your git repo file in that case the reports about `name` and `version` don't make sense --- could you maybe check whether you are in a VCS before reporting them (maybe it should also say that this was a repo validation) ? Example:

```
> opam lint 
Validation failed for ~/sync/repos/xmlm/opam: 
  - Field 'opam-version' doesn't match the current version, validation may not be accurate
  - Missing field 'name' or directory in the form 'name.version'
  - Missing field 'version' or directory in the form 'name.version'
  - Field 'ocaml-version' is deprecated, use 'available' instead
  - Missing field 'bug-reports'
  - Missing field 'dev-repo'
```

Also I don't know if this a UK vs US english issue and whether one was choosen for `opam` but I'm used to "normalize" in english, the option is called `--normalise`.

Finally is there any reason why it's not possible to perform a best-effort normalization; report warnings rather than errors and continue as far as possible: 

```
> opam lint --normalise
Validation failed for ~/sync/repos/xmlm/opam: 
  - Field 'opam-version' doesn't match the current version, validation may not be accurate
  - Missing field 'name' or directory in the form 'name.version'
  - Missing field 'version' or directory in the form 'name.version'
  - Field 'ocaml-version' is deprecated, use 'available' instead
  - Missing field 'bug-reports'
  - Missing field 'dev-repo'
[ERROR] Invalid OPAM file (missing field "version")
```

--- a/src/core/opamFile.ml
+++ b/src/core/opamFile.ml
@@ -1423,10 +1423,12 @@ module X = struct
                 <> 0)
             "Field 'opam-version' doesn't match the current version, validation \
              may not be accurate";
+(*
           cond (t.name = None)
             "Missing field 'name' or directory in the form 'name.version'";
           cond (t.version = None)
             "Missing field 'version' or directory in the form 'name.version'";
+*)
           cond (t.maintainer = [""] || t.homepage = [""] || t.author = [""] ||
                 t.license = [""] || t.doc = [""] || t.tags = [""] ||
                 t.bug_reports = [""])

